Src/Fido2/AttestationFormat/AppleAppAttest.cs

```diff
@@ -81,22 +81,20 @@ public override (AttestationType, X509Certificate2[]) Verify(VerifyAttestationRe
         // 4. Obtain the value of the credCert extension with OID 1.2.840.113635.100.8.2, which is a DER - encoded ASN.1 sequence.Decode the sequence and extract the single octet string that it contains. Verify that the string equals nonce.
         // Steps 2 - 4 done in the "apple" format verifier
         var apple = new Apple();
-        (var attType, var trustPath) = apple.Verify(request);
+        (var attType, var trustPath) = await apple.VerifyAsync(request).ConfigureAwait(false);
 
         // 5. Create the SHA256 hash of the public key in credCert, and verify that it matches the key identifier from your app.
-        Span<byte> credCertPKHash = stackalloc byte[32];
-        SHA256.HashData(credCert.GetPublicKey(), credCertPKHash);
-        ReadOnlySpan<byte> keyIdentifier = Convert.FromHexString(credCert.GetNameInfo(X509NameType.SimpleName, false));
-        if (!credCertPKHash.SequenceEqual(keyIdentifier))
+        byte[] credCertPKHash = SHA256.HashData(credCert.GetPublicKey());
+        byte[] keyIdentifier = Convert.FromHexString(credCert.GetNameInfo(X509NameType.SimpleName, false));
+        if (!credCertPKHash.AsSpan().SequenceEqual(keyIdentifier))
         {
             throw new Fido2VerificationException("Public key hash does not match key identifier in Apple AppAttest attestation");
         }
 
         // 6. Compute the SHA256 hash of your app's App ID, and verify that it’s the same as the authenticator data's RP ID hash.
         var appId = GetAppleAppIdFromCredCertExtValue(credCert.Extensions);
-        Span<byte> appIdHash = stackalloc byte[32];
-        SHA256.HashData(appId, appIdHash);
-        if (!appIdHash.SequenceEqual(request.AuthData.RpIdHash))
+        byte[] appIdHash = SHA256.HashData(appId);
```

Comment:
I'm not sure to understand this change. Why not continue to hash into a `stackalloc byte[32]` directly instead of allocating a byte array?

Reply:
ref structs (including Spans) aren't allowed inside of an async block.

Reply:
Right, ok!